pontoon/sync/formats/compare_locales.py

```diff
@@ -0,0 +1,116 @@
+"""
+Parser and serializer for file formats supported by compare-locales library:
+https://hg.mozilla.org/l10n/compare-locales/
+"""
+from __future__ import absolute_import
+
+import copy
+import logging
+
+from collections import OrderedDict
+from compare_locales import (
+    parser,
+    serializer,
+)
+
+from pontoon.sync import SyncError
+from pontoon.sync.exceptions import ParseError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+
+log = logging.getLogger(__name__)
+
+
+class CompareLocalesEntity(VCSTranslation):
+    """
+    Represents an entity in a file handled by compare-locales.
+    """
+    def __init__(self, key, string, comment, order):
+        self.key = key
+        self.source_string = string
+        self.source_string_plural = ''
+        self.strings = {None: self.source_string} if self.source_string else {}
+        self.comments = [comment.raw_val.strip()] if comment else []
+        self.order = order
+        self.fuzzy = False
+        self.source = []
+
+
+class CompareLocalesResource(ParsedResource):
+    def __init__(self, path, source_resource=None):
+        self.path = path
+        self.entities = OrderedDict()  # Preserve entity order.
+        self.source_resource = source_resource
+        self.parser = copy.deepcopy(parser.getParser(self.path))
```

Comment:
We need to `deepcopy` here or else parser contexts from l10n resource and reference resource override each other.

We should add a comment, but I hope there's a better way of implementing this. :)